src/pages/api/addList.ts

```diff
@@ -0,0 +1,31 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { z, ZodError } from "zod";
+import { Items, List, Prisma, PrismaClient } from "@prisma/client";
+
+const prisma = new PrismaClient();
+
+export default async function handler(
+  request: NextApiRequest,
+  response: NextApiResponse
+) {
+  if (request.method === "POST") {
+    const { name, user } = inputQueryTest.parse(request.body);
+
+    const listToBe = await prisma.list.create({
+      data: {
+        name: name,
+        items: { create: [] },
+        user: { connect: { id: user } },
+      },
+    });
+
+    response.status(200).json({ message: "List added successfully" });
+  } else {
+    response.status(405).json({ message: "Method not allowed" });
```

Comment:
if you're already adding error handling, why not try catch the prisma client and handle Zod validation errors?

Reply:
changed with 'try' and working by testing with htt page, new list message 200 is there and appearing on prisma studio.